raiden-ts/src/raiden.ts

```diff
@@ -700,35 +728,82 @@ export class Raiden {
     if (!UInt(32).is(amount)) throw new Error('Invalid amount');
 
     const paymentId = !options.paymentId ? undefined : bigNumberify(options.paymentId);
-    if (paymentId && !UInt(8).is(paymentId)) throw new Error('Invalid opts.paymentId');
+    if (paymentId && !UInt(8).is(paymentId)) throw new Error('Invalid options.paymentId');
 
     if (options.secret !== undefined && !Secret.is(options.secret))
-      throw new Error('Invalid opts.secret');
+      throw new Error('Invalid options.secret');
     if (options.secrethash !== undefined && !Hash.is(options.secrethash))
-      throw new Error('Invalid opts.secrethash');
-    let { secret, secrethash } = options;
-    if (!secrethash) {
-      if (!secret) secret = makeSecret();
-      secrethash = getSecrethash(secret);
-    } else if (secret && getSecrethash(secret) !== secrethash) {
-      throw new Error('Secret and secrethash must match if passing both');
-    }
-
-    const promise = this.action$
-      .pipe(
-        filter(isActionOf([transferSigned, transferFailed])),
-        first(action => action.meta.secrethash === secrethash!),
+      throw new Error('Invalid options.secrethash');
+
+    // use provided secret or create one if no secrethash was provided
+    const secret = options.secret
+        ? options.secret
+        : !options.secrethash
+        ? makeSecret()
+        : undefined,
+      secrethash = options.secrethash || getSecrethash(secret!);
+    if (secret && getSecrethash(secret) !== secrethash)
+      throw new Error('Provided secrethash must match the sha256 hash of provided secret');
+
+    const metadata = options.metadata;
+    if (metadata && !Metadata.is(metadata)) throw new Error('Invalid options.metadata');
+
+    return merge(
+      // wait for pathFind response
+      this.action$.pipe(
+        filter(isActionOf([pathFound, pathFindFailed])),
+        first(
+          action =>
+            action.meta.tokenNetwork === tokenNetwork &&
+            action.meta.target === target &&
+            action.meta.value.eq(amount),
+        ),
+        timeout(10e3), // timeout if no corresponding success/error response before 10s
```

Comment:
Might make sense to have an `httpTimeout` in `defaultConfig` and also give devs the options to define their own timeout

Reply:
Ok, but then it doesn't go here, but inside the PFS epic itself. Will do

Reply:
Done, included an `httpTimeout` config option